ovs/datapath_test.go

```diff
@@ -0,0 +1,656 @@
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: uint8(10),
+				},
+			},
+			dpName: "test-datapath",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			err := tt.dp.DelDataPath(tt.dpName)
+			if err != nil {
+				t.Errorf("getting an error while trying to get version %q", err.Error())
+			}
+		})
+	}
+}
+
+func TestGetCTLimits(t *testing.T) {
+	z0 := make(Zone)
+	z0["default"] = 0
+	z1 := make(Zone)
+	z1["zone"] = 2
+	z1["count"] = 0
+	z1["limit"] = 1000000
+	z2 := make(Zone)
+	z2["zone"] = 3
+	z2["count"] = 0
+	z2["limit"] = 1000000
+	zones := make([]Zone, 0)
+	zones = append(zones, z1)
+	zones = append(zones, z2)
+	out := &ConnTrackOutput{
+		defaultLimit: z0,
+		zones:        zones,
+	}
+
+	var tests = []struct {
+		desc     string
+		dp       *DataPathService
+		dpName   string
+		zones    []uint64
+		want     *ConnTrackOutput
+		err      string
+		testCase uint8
+	}{
+		{
+			desc: "Test invalid ovs-dpctl version ",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: unknownCommand,
+				},
+			},
+			dpName:   "ovs-system",
+			zones:    []uint64{2, 3},
+			want:     nil,
+			err:      "ovs-dpctl: unknown command 'ct-get-limits'; use --help for help",
+			testCase: unknownCommand,
+		},
+		{
+			desc: "Test valid ovs-dpctl version ",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 10,
+				},
+			},
+			dpName:   "ovs-system",
+			zones:    []uint64{2, 3},
+			want:     out,
+			err:      "",
+			testCase: validTest,
+		},
+		{
+			desc: "Test valid ovs-dpctl ct-get-limits",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 10,
+				},
+			},
+			dpName:   "",
+			zones:    []uint64{},
+			want:     out,
+			err:      "datapath name argument is mandatory",
+			testCase: handleError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			results, err := tt.dp.GetCTLimits(tt.dpName, tt.zones)
+			switch tt.testCase {
+			case unknownCommand:
+				if err.Error() != tt.err {
+					t.Errorf("getting an error while trying to get conntrack limits %q", err.Error())
+				}
+			case validTest:
+				if tt.want.defaultLimit["default"] != results.defaultLimit["default"] {
+					t.Errorf("mismatched values, want %q  but got %q", tt.want.defaultLimit["default"], results.defaultLimit["default"])
+				}
+				for i, z := range results.zones {
+					if tt.want.zones[i]["zone"] != z["zone"] {
+						t.Errorf("mismatched values, want %q  but got %q", tt.want.zones[i]["zone"], z["zone"])
+					}
+				}
+			case handleError:
+				if err != nil && err.Error() != tt.err {
+					t.Errorf(err.Error())
+				}
+			default:
+				t.Log("pass")
+			}
+		})
+	}
+}
+
+func TestGetCTLimitsWithBinary(t *testing.T) {
+	if testing.Short() {
+		// if you want to run tests in a context where
+		// you don't get ovs-dpctl installed, you should run
+		// go test -short to skip it
+		t.Skip("skipping test in short mode.")
+	}
+
+	z0 := make(Zone)
```

Comment:
nit: here it might be confusing since z0 doesn't represent zone 0 but the default limit